core/src/main/scala/org/apache/spark/sql/rapids/tool/store/StageModel.scala

```diff
@@ -27,11 +27,33 @@ import org.apache.spark.sql.rapids.tool.annotation.{Calculated, Since, WallClock
  *
  * @param sInfo Snapshot from the stage info loaded from the eventlog
  */
+
 @Since("24.02.3")
-class StageModel private(var sInfo: StageInfo) {
+class StageModel private(sInfo: StageInfo) {
 
-  val sId: Int = sInfo.stageId
-  val attemptId: Int = sInfo.attemptNumber()
+  var stageInfo: StageInfo = initStageInfo(sInfo)
```

Comment:
One thing we can do here is to update updateInfo() in the class body so it initializes the stageInfo.

```
var stageInfo: StageInfo =  _
//...
updateInfo(sInfo)

```

Reply:
Ahmed, in that case should be remove the init as a function separately and include that code in update. Or you would prefer keeping the functionality separate ? Because init is just creating a new StateInfo object